pkg/api/types/field.go

```diff
@@ -9,26 +9,26 @@ import (
 // A FieldDefinition defines the name, description, boundaries of data collection.
 type FieldDefinition struct {
 	// ID is the ID of the FieldDefinition
-	ID string `json:"id"`
+	ID string `json:"id,omitempty" yaml:"id,omitempty"`
```

Comment:
I don't think we should be setting `omitempty` on all of these fields. Most of them seem to be required fields and should error during read/writes validation. By adding `omitempty` it indicates that a missing value is an expected state and not an error.

Reply:
When a user submits a `FormDefinition` with `FieldDefinitions`, in this case the `id` is supposed to be empty, since this generated (server side uuids)...

Reply:
Ah yes, so in that case I don't think a `FieldDefinition` should have and `ID`. There should be a different struct for when we read from the database.

But I think my point stands for the other fields.

Reply:
Just modified for the "required" fields to not have `omitempty`
I think if we have different structs for read & write, this would be another PR entirely, and would be quite big

Reply:
Yeah, not a suggestion for this pr but I think it's a better pattern generally.